third_party/txt/src/txt/paragraph_txt.cc

```diff
@@ -1030,7 +1036,13 @@ void ParagraphTxt::Layout(double width) {
                               ? glyph_positions.back().x_pos.start +
                                     run.placeholder_run()->width
                               : glyph_positions.back().x_pos.end),
-            line_number, metrics, run.direction(), run.placeholder_run());
+            line_number, metrics, run.style(), run.direction(),
+            run.placeholder_run());
+
+        line_metrics.run_metrics_map.emplace(
+            std::piecewise_construct, std::forward_as_tuple(run.start()),
+            std::forward_as_tuple(run.style(), metrics));
```

Comment:
do we pay for this copy every time we paint text, even when we never look at the metrics? (or is it cheaper than a copy?)